_use_cases/encryption-at-rest.markdown

```diff
@@ -0,0 +1,105 @@
+data is now residing in the clear on the file systems of the service provider.
+
+| ![image](../assets/encryption-at-rest_problem.png){:width="100%"} |
+|:-----------------------------------------------------------------:|
+|    *Problem: Plain text records readable by the Kafka Admins*     |
+
+### Isn't TLS sufficient?
+
+TLS encrypts the content _on the wire_.  It means that someone using a network sniffer cannot tap what is being
+sent over the wire between the application and the Kafka Broker.  However, once the network packets arrive at the broker,
+the packets are decrypted and exist in the clear once again.  This means the confidential records are in the clear in the memory
+of the broker and in the clear when the data is written to the file system.
+
+TLS does not change the problem.
+
+### Isn't storage volume encryption an answer?
+
+With storage volume encryption, the contents of the device are encrypted with a key.  This approach provides some mitigations.
+If the storage device is stolen or the storage device hijacked and attached to an attacker's computer, the attacker won't have
+the encryption key so won't be able to read the data, including the Kafka records.
+
+However, there are shortcomings.  The encrypted storage volume must be mounted to the computers running the Kafka broker. To do
+this the  and the encryption keys applied at the operating system level.  Anybody with shell level access to the hosts  is likely
+to be able to read the data, including the Kafka confidential records.
+
+Storage volume encryption doesn't really solve the problem.
+
+### Can't the applications encrypt/decrypt the data?
+
+It is possible for producing applications to encrypt data before sending it to Kafka, and for consuming applications to decrypt it
+again.  With this approach the brokers never possess the records in the clear and as they don't have encryption keys, they cannot
+decrypt it.
+
+So, this approach does offer a solution to the problem but there are disadvantages.
+
+1. Kafka Client libraries don't support encryption themselves.
+1. Kafka Client ecosystem is polyglot - it is common to Kafka applications written in many languages (Java, Go, Rust...)
+   deployed within a _single organisation_.  Any solution needs to work across all the target languages.
+1. There are some encryption libraries for some languages that can be used with the Kafka Clients but these available
+   across all languages.
+
+Organisations could write their own encryption code but this is a burden to the application teams. The need for
+_interoperability_ between different language encryption implementations - this increases the overheads.  There is also
+the problem of key distribution to consider.  Some part of the system needs to push the correct encryption keys out to
+the applications and manage tasks such as key rotation.  Finally, cryptography is a specialist area and the consequences
+of a design flaw or bug are significant (confidentiality breach).
```

Comment:
Spell out the security risks of having to push keys to many more places (applications) than in a more centralised approach.

Other points to make:

*You say it's specialist and comes with consequences, but spell out how that combines with the polyglot point to push up the costs of doing this correctly in multiple places.
* You can also point out that even with support in client libraries and languages it might not be enough because any 3rd party tooling and applications won't use that stuff. 
* Finally, these might be acceptable costs _today_, but the costs can increase later on (e.g. we now want to use this new tool/client, but can't).